pkg/awsutils/awssession/session.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package awssession
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"strconv"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/utils/logger"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+// Http client timeout env for sessions
+const (
+	httpTimeoutEnv = "HTTP_TIMEOUT"
+	maxRetries     = 10
+)
+
+var (
+	version string
+	log     = logger.Get()
+	// HTTP timeout default value in seconds (10 seconds)
+	httpTimeoutValue = 10 * time.Second
+)
+
+func getHTTPTimeout() time.Duration {
+	httpTimeoutEnvInput := os.Getenv(httpTimeoutEnv)
+	// if httpTimeout is not empty, we convert value to int and overwrite default httpTimeoutValue
+	if httpTimeoutEnvInput != "" {
+		input, err := strconv.Atoi(httpTimeoutEnvInput)
+		if err == nil && input >= 10 {
+			log.Debugf("Using HTTP_TIMEOUT %v", input)
+			httpTimeoutValue = time.Duration(input) * time.Second
+			return httpTimeoutValue
+		}
+	}
+	log.Warn("HTTP_TIMEOUT env is not set or set to less than 10 seconds, defaulting to httpTimeout to 10sec")
```

Comment:
log.Warn -> log.Info 

Nothing to warn about as most users will just go with the default. Consider logging a warning if the env variable is set and the value falls out of bounds or is invalid otherwise (i.e not a valid integar)